packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionsTableDetailsSteps.tsx

```diff
@@ -0,0 +1,180 @@
+
+function getTransferDurationText(tx: MergedTransaction) {
+  const { isTestnet, isOrbitChain } = isNetwork(tx.parentChainId)
+
+  if (tx.isCctp) {
+    return isTestnet ? '1 minute' : '10 minutes'
+  }
+
+  if (!tx.isWithdrawal) {
+    if (isOrbitChain) {
+      return '1 minute'
+    }
+    return isTestnet ? '10 minutes' : '15 minutes'
+  }
+
+  // withdrawals
+  return getTxConfirmationDate({
+    createdAt: dayjs(),
+    withdrawalFromChainId: tx.childChainId
+    // we set from to current time so that we get the full withdrawal confirmation time
+  }).from(dayjs(), true)
+}
+
+function needsToClaimTransfer(tx: MergedTransaction) {
+  return tx.isCctp || tx.isWithdrawal
+}
+
+const Step = ({
+  done = false,
+  claimable = false,
+  pending = false,
+  text,
+  endItem = null
+}: {
+  done?: boolean
+  claimable?: boolean
+  pending?: boolean
+  text: string
+  endItem?: ReactNode
+}) => {
+  // defaults to a step that hasn't been started yet
+  let borderColorClassName = 'border-white/50'
+  let textColorClassName = 'text-white/50'
+
+  if (done || claimable) {
+    borderColorClassName = 'border-green-400'
+    textColorClassName = 'text-green-400'
+  }
+
+  if (pending) {
+    borderColorClassName = 'border-yellow-400'
+    textColorClassName = 'text-yellow-400'
+  }
+
+  return (
+    <div
+      className={twMerge(
+        'my-3 flex items-center justify-between',
+        pending ? 'animate-pulse' : '',
+        claimable ? 'my-2' : '',
+        textColorClassName
+      )}
+    >
+      <div className="flex items-center space-x-3">
+        {done ? (
+          <CheckCircleIcon height={18} />
+        ) : (
+          <div
+            className={twMerge(
+              'ml-[2px] h-[15px] w-[15px] rounded-full border',
+              borderColorClassName
+            )}
+          />
+        )}
+        <span>{text}</span>
+      </div>
+      {endItem}
+    </div>
+  )
+}
+
+export const TransactionsTableDetailsSteps = ({
+  tx
+}: {
+  tx: MergedTransaction
+}) => {
+  const { parentLayer, layer: childLayer } = useChainLayers({
+    parentChainIdOverride: tx.parentChainId,
+    childChainIdOverride: tx.childChainId
+  })
+  const { remainingTime: cctpRemainingTime } = useRemainingTime(tx)
+
+  return (
+    <div className="flex flex-col text-xs">
+      {/* First step when transfer is initiated */}
+      <Step
+        done
+        text={`Transaction initiated on ${
+          tx.isWithdrawal ? childLayer : parentLayer
+        }`}
+        endItem={
+          <ExternalLink
+            href={`${getExplorerUrl(
+              tx.isWithdrawal ? tx.childChainId : tx.parentChainId
+            )}/tx/${tx.txId}`}
+          >
+            <ArrowTopRightOnSquareIcon height={12} />
+          </ExternalLink>
+        }
+      />
+
+      {/* Pending transfer showing the remaining time */}
+      <Step
+        pending={isTxPending(tx)}
+        done={!isTxPending(tx)}
+        text={`Wait ${getTransferDurationText(tx)}`}
```

Comment:
We don't know if it will take exactly the time we state. Should add `about` or `approximately` here.

```suggestion
        text={`Wait for about ${getTransferDurationText(tx)}`}
```

Reply:
I'll do `Wait ~10 minutes`, I think less text is nicer